### PR TITLE
Fix recycled Bitmap check.

### PR DESCRIPTION
**Pull request by NightlyNexus**
fixes the sample app crashing at start.

### CODE REVIEW

File: picasso/src/main/java/com/squareup/picasso3/BitmapHunter.java [hunk 1]
```diff
@@ -428,7 +428,9 @@ static Result applyTransformations(Picasso picasso, Request data,
       }
 
       // If the transformation returned a new bitmap ensure they recycled the original.
-      if (newResult != result && !bitmap.isRecycled()) {
+      if (newResult != result
+          && newResult.getBitmap() != result.getBitmap()
```
Comment (JakeWharton) [inline picasso/src/main/java/com/squareup/picasso3/BitmapHunter.java]: Left side can just be bitmap, right?

Comment (NightlyNexus) [reply picasso/src/main/java/com/squareup/picasso3/BitmapHunter.java]: right side.

**Comment by NightlyNexus** [comment]
yeah, the matrix transformation. must have been broken before, but no transformations are in the sample app.

**Comment by jrodbx** [comment]
Interesting, the crash seems to only happen on P.  Tested on a Nexus 5X API 25, Pixel 2 API 27, and a Nexus 5X API 28.

**Pull request merged**